src/client/components/chart-detail-page-layout/chart-detail-page.component.js

```diff
@@ -0,0 +1,91 @@
+import React, { useContext } from 'react';
+import DetailChart from '../detail-chart/detail-chart.component';
+import SidebarMenu from '../side-navigation/sidebar.component';
+import ProgressBar from '../progress-bar/progress-bar.component';
+import './chart-detail-page.css';
+import ChartbarMenu from '../chart-data-buttons/chartbar-buttons/chart-data-button.component';
+import { ChartDataContext } from '../../containers/chart-detail-page/chart-detail-page.context';
+import UserRoleContext from '../../helpers/UserRoleContext';
+import { useHistory } from 'react-router-dom';
+import Logout from '../logout/logout.component';
+import Firebase from '../../firebase/index';
+
+const dateButtons = [
+  { id: 1, label: 'Last 5 Days' },
+  { id: 2, label: 'Last Week' },
+  { id: 3, label: 'Last Two Weeks' },
+  { id: 4, label: 'Last Month' },
+  { id: 5, label: 'Custom' },
+];
+
+const ChartDetailPage = () => {
+  const history = useHistory();
+  const {
+    boundaryData,
+    materialName,
+    sensorData,
+    startDate,
+    currentDate,
+    stages,
+    unit,
+    selectedChartButtonId,
+    setSelectedChartButtonId,
+    activeChartButton,
+    setActiveChartButton,
+    setStartCustom,
+    setEndCustom,
+    updateClick,
+    setUpdateClick,
+    logoutModal,
+    setLogoutModal,
+  } = useContext(ChartDataContext);
+  const { userRole, userName } = useContext(UserRoleContext);
+  const headingText = materialName.toUpperCase();
+  return (
+    <>
+      <SidebarMenu
```

Comment:
Ok, apparently you are replicating the entire sidebar menu and other layout items on each page. We can definitely re-use that better, but again I think we should do it outside this PR.